sys/net/link_layer/ieee802154/submac.c

```diff
@@ -285,8 +287,8 @@ int ieee802154_submac_init(ieee802154_submac_t *submac)
     ieee802154_radio_request_on(dev);
 
     /* generate EUI-64 and short address */
-    luid_get_eui64(&submac->ext_addr);
-    luid_get_short(&submac->short_addr);
+    netdev_eui64_get(&netdev->dev.netdev, &submac->ext_addr);
```

Comment:
In order to remove netdev dependencies from the SubMAC, do you think this could be possible in `netdev_ieee802154_submac`? We could either change the `ieee802154_submac_init` signature to receive these address, or simply call the address setters right after init (in `netdev_ieee802154_submac_init`).

What do you think?

Reply:
You mean remove `netdev_t` entirely? 
Will there anything common about different network devices then anymore?

Reply:
e.g somewhere around here:
https://github.com/RIOT-OS/RIOT/blob/6cb8da7813b1250d53ea7fd4ce54164899d89bd7/drivers/netdev_ieee802154_submac/netdev_ieee802154_submac.c#L284

Reply:
Why is there a copy of ext_addr / short_addr *both* in `ieee802154_submac_t` and `netdev_ieee802154_t`?

Reply:
> Will there anything common about different network devices then anymore?

I think so. 
But the problem is, `netdev_t` defines the `netdev_driver_t` structure, which is not used anymore.
I think we could use "netdev" as a generic device descriptor, but I think we should be careful to avoid the confusion between "netdev as a generic device descriptor" and "netdev as a network device interface"

Reply:
> Why is there a copy of ext_addr / short_addr both in ieee802154_submac_t and netdev_ieee802154_t?

Because the netdev components were intended to be used as a transition layer, not as an end solution. The idea is to remove netdev from IEEE 802.15.4 (at least as a "device interface"). I agree we need a "descriptor", but I think we should start opening the discussion now of which level should this descriptor cover (network interface? device level? etc)